proposals/0029-cooperative-vector.md

```diff
@@ -331,36 +331,98 @@ can be found in [Minimum Support Set].
 
 ### Type Interpretations
 
-The various "interpretation" arguments specify a value from the following enum:
+The `ComponentType` enum in `DxilConstants.h` is extended as shown below, with
+four new 8-bit types:
 
 ```c++
-enum class DXILTypeInterpretation :uint {
-  Float16               = 0,
-  Float32               = 1,
-  UnsignedInt8          = 2,
-  UnsignedInt16         = 3,
-  UnsignedInt32         = 4,
-  SignedInt8            = 5,
-  SignedInt16           = 6,
-  SignedInt32           = 7,
-  SignedInt8x4Packed    = 8,
-  UnsignedInt8x4Packed  = 9,
-  FloatE4M3             = 10,
-  FloatE5M2             = 11,
-  Unsupported           = 32
+enum class ComponentType : uint32_t {
+  Invalid = 0,
+  I1,
+  I16, // = 2
+  U16, // = 3
+  I32, // = 4
+  U32, // = 5
+  I64,
+  U64,
+  F16, // = 7
+  F32, // = 8
+  F64,
+  SNormF16,
+  UNormF16,
+  SNormF32,
+  UNormF32,
+  SNormF64,
+  UNormF64,
+  PackedS8x32, // = 16
+  PackedU8x32, // = 17
+
+  // BEGIN NEW FOR SM 6.9
+  U8,      // = 18
+  I8,      // = 19
+  F8_E4M3, // = 20  
+  F8_E5M2, // = 21
+  // END     
+
+  LastEntry
 };
 ```
 
-For matrices and vectors that are specified by resource handles and stored in
-raw-buffers, the interpretation value directly specifies the element type.  It
-is invalid to specify a packed interpretation in these cases.
-
-For input vectors that come from variables there is a distinction between the
-physical type and the logical type. The **input interpretation** argument for
-these vectors describes how to convert from the physical to logical type. This
-allows elements to be interpreted as types not natively supported by HLSL, e.g.
-uint8/sint8. For packed interpretations, a single physical element can expand
-into multiple logical elements.
+#### From-Register Interpretations
+
+Input vectors stored in registers (eg `vector<float, 16>`) are interpreted
+according to the Conversion Rules shown below.
+
+For these vectors there is a distinction between the physical type and the
+logical type. The **input interpretation** argument for these vectors describes
+how to convert from the physical to logical type. This allows elements to be
+interpreted as types not natively supported by HLSL, e.g. uint8/sint8. For
+packed interpretations, a single physical element can expand into multiple
+logical elements.
```

Comment:
Would there ever be a reason to do multiple physical elements to a single logical element? Just curious.

Reply:
Maybe for 64-bit types, but the trend is for less bits-per-element!